src/fstar/FStar.Interactive.Ide.fs

```diff
@@ -98,11 +60,11 @@ let with_captured_errors' env sigint_handler f =
     Util.print_string "Interrupted"; None
 
   | Error (e, msg, r) ->
-    TcErr.add_errors env [(e, msg, r)];
+    Errors.add_errors [(e, msg, r)];
```

Comment:
Why is this change required?

Reply:
To be honest, I don't remember why I made that change. Whatever the original reason was, it no longer seems to be valid, since reverting the change doesn't break anything -- thanks a lot of catching this!

Reply:
I'm very sorry about this, but this change is required to make the Ide regression suite pass — I wish I could remember how the change ties in, but I can't; it's been almost 2 months since I posted this branch for review, and I can scarcely remember the details 😓

Reply:
I changed tc_one_fragment to read the filename from the env instead of command-line options (https://github.com/FStarLang/FStar/pull/1744/files#diff-0f8be5d4810e22817ca655ca5b7778feR141), because we don’t provide a command-line option in the case of Lsp. Now, the filename in the env is “<input>” in Ide because that’s what init_range used to set (https://github.com/FStarLang/FStar/pull/1744/files#diff-324a3d5d7227d9b71dbb0baba0fce7a5R1081) — however, if we change this, then the error reporting in Ide needs to be tweaked slightly to not print an elaborate message when the fname in env and repl_fname are different (https://github.com/FStarLang/FStar/pull/1744/files#diff-324a3d5d7227d9b71dbb0baba0fce7a5R63). This change to Ide.fs is hence unavoidable.